### PR TITLE
Cleaning up JArchiveBzip2

### PR DESCRIPTION
**Pull request by Mathewlenning**
Isolated JError into temporary method and separated old extraction logic from streaming logic.

I'm not sure where this is used, but most likely in the installer. 
I did a find usage with PHPStorm, but the only results I got where from the unit tests. 

So if anyone knows where it is used could you let me know.

### CODE REVIEW

**Comment by Mathewlenning** [comment]
For some reason this didn't get tagged and TCI wasn't initiated. Is Bzip2 on the exclude list?

**Comment by wilsonge** [comment]
Nothing should be on the exclude list. I'll try and work out what's going on!

**Comment by Mathewlenning** [comment]
It started up after I pushed a few style fixes. I'm guessing it was just a hiccup in TCI

**Comment by Kubik-Rubik** [comment]
Looks good to me. We won't add this in the 3.4.1 but after another successful test we can merge it for the 3.4.2 version. Thank you @Mathewlenning!

File: libraries/joomla/archive/bzip2.php [hunk 1]
```diff
@@ -45,125 +45,108 @@ public function extract($archive, $destination, array $options = array ())
...
-				{
-					return JError::raiseWarning(100, 'Unable to decompress data');
-				}
-				else
-				{
-					throw new RuntimeException('Unable to decompress data');
-				}
-			}
+		$buffer = bzdecompress($this->_data);
+		unset($this->_data);
```
Comment (Hackwar) [inline libraries/joomla/archive/bzip2.php]: Why do you unset this here? Shouldn't this be set to null or something?

Comment (Mathewlenning) [inline libraries/joomla/archive/bzip2.php]: That is the original behavior. Changing an established side effect without a good reason doesn't make sense. Setting it to null has no benefit to cleanliness or clarity, but it does present the possibility of breaking something if someone was expecting it to be unset.

**Comment by Kubik-Rubik** [comment]
Thank you @Mathewlenning!

**Pull request merged**